plugins/kafka/CMakeLists.txt

```diff
@@ -0,0 +1,113 @@
+# Description:
+# ------------
+#    Cmake Input File for kafka
+#####################################################
+
+project( kafka )
+
+if (USE_KAFKA)
+
+    ADD_PLUGIN(kafka PACKAGES OPTION MAKE_KAFKA)
+
+    if ( MAKE_KAFKA )
+
+        # librdkafka packages do not include everything we need to properly
+        # build against it; until/if they are ever fixed we will need to build
+        # our own version of librdkafka from source and include it ourselves
+
+        if( NOT EXISTS "${PROJECT_SOURCE_DIR}/librdkafka/configure" )
+            message( FATAL_ERROR
+"   The librdkafka submodule is not available.
+   This normally indicates that the git submodule has not been fetched.
+   Please run git submodule update --init --recursive")
+        endif()
+
+        # librdkafka does not support out-of-source builds, so let's copy all
+        # of its source code to our binary directory and build there; further,
+        # we need to pull some working directory shenanigans for each command
+        # in order to make the built scripts function correctly
+
+        add_custom_command ( OUTPUT ${PROJECT_BINARY_DIR}/lib/librdkafka.a
+                COMMAND cp -r ${PROJECT_SOURCE_DIR}/librdkafka ${PROJECT_BINARY_DIR}/src
+                COMMAND cd ${PROJECT_BINARY_DIR}/src && ./configure --prefix=${PROJECT_BINARY_DIR}
+                COMMAND cd ${PROJECT_BINARY_DIR}/src && make && make install
+                COMMENT Copying and building librdkafka
+            )
+
+        add_custom_target ( librdkafka-build ALL DEPENDS ${PROJECT_BINARY_DIR}/lib/librdkafka.a )
+
+        # Add both static libraries from librdkafka
+
+        add_library ( librdkafka STATIC IMPORTED )
+        set_property ( TARGET librdkafka PROPERTY IMPORTED_LOCATION ${PROJECT_BINARY_DIR}/lib/librdkafka.a )
+        add_dependencies ( librdkafka librdkafka-build )
+
+        add_library ( librdkafkacpp STATIC IMPORTED )
+        set_property ( TARGET librdkafkacpp PROPERTY IMPORTED_LOCATION ${PROJECT_BINARY_DIR}/lib/librdkafka++.a )
+        add_dependencies ( librdkafkacpp librdkafka-build )
+
+        set (   SRCS
+                kafka.hpp
+                kafka.cpp
+            )
+
+        include_directories (
+                ./../../system/include
+                ./../../rtl/eclrtl
+                ./../../rtl/include
+                ./../../common/deftype
+                ./../../system/jlib
+                ${PROJECT_BINARY_DIR}/include
+                ${CMAKE_BINARY_DIR}
+            )
+
+        ADD_DEFINITIONS( -D_USRDLL -DECL_KAFKA_EXPORTS )
+        HPCC_ADD_LIBRARY( kafka SHARED ${SRCS} )
+
+        if (${CMAKE_VERSION} VERSION_LESS "2.8.9")
+            message("WARNING: Cannot set NO_SONAME. shlibdeps will give warnings when package is installed")
+        elseif(NOT APPLE)
+            set_target_properties( kafka PROPERTIES NO_SONAME 1 )
+        endif()
+
+        install ( TARGETS kafka DESTINATION plugins )
+
+        if (APPLE)
```

Comment:
I targeted OS X specifically because the problem seemed to lie only with that one platform. It may be overkill to supply them for all platforms, but it wouldn't hurt.  Do you recommend adding zlibs unconditionally?